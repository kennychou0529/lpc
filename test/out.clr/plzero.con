
compiling pass 1 (LPC)
-----------------------------------------------------
  warning on line 66: non-local goto
  warning on line 115: non-local goto
  warning on line 117: 'a' hiding existing declaration on line 42
  warning on line 151: 'id' hiding existing declaration on line 35
  warning on line 183: 'a' hiding existing declaration on line 42
  warning on line 187: 'fsys' hiding existing declaration on line 130
  warning on line 189: 'fsys' hiding existing declaration on line 187
  warning on line 189: 'fsys' hiding existing declaration on line 130
  warning on line 191: 'fsys' hiding existing declaration on line 189
  warning on line 191: 'fsys' hiding existing declaration on line 187
  warning on line 191: 'fsys' hiding existing declaration on line 130
  warning on line 193: 'fsys' hiding existing declaration on line 191
  warning on line 193: 'fsys' hiding existing declaration on line 189
  warning on line 193: 'fsys' hiding existing declaration on line 187
  warning on line 193: 'fsys' hiding existing declaration on line 130
  warning on line 194: 'i' hiding existing declaration on line 188
  warning on line 240: 'fsys' hiding existing declaration on line 187
  warning on line 240: 'fsys' hiding existing declaration on line 130
  warning on line 374: 'a' hiding existing declaration on line 42

0 error(s), 19 warning(s)

Exit code: 0

compiling pass 2 (target = clr)
-----------------------------------------------------

Invoking ILASM ...
Exit code: 0

Verifying the compiled binary (PEVERIFY) ...
[IL]: Error: [plzero.exe : <Module>::_SetContains][offset 0x00000005][found address of value 'T_set_2'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : <Module>::_SetContains][offset 0x00000004][found address of value 'T_symset'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : <Module>::_SetCastTo_T_set_2][offset 0x00000005] Instruction cannot be verified.
[IL]: Error: [plzero.exe : <Module>::_SetUnion][offset 0x00000008][found address of value 'T_set_2'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : <Module>::_SetUnion][offset 0x00000008][found address of value 'T_set_2'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : <Module>::_SetUnion][offset 0x00000008][found address of value 'T_set_2'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : <Module>::_SetCastTo_T_symset][offset 0x00000005] Instruction cannot be verified.
[IL]: Error: [plzero.exe : <Module>::_SetUnion][offset 0x00000007][found address of value 'T_symset'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : <Module>::_SetUnion][offset 0x00000007][found address of value 'T_symset'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : <Module>::_SetUnion][offset 0x00000007][found address of value 'T_symset'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : <Module>::_SetCheck][offset 0x00000006][found address of value 'T_set_2'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : T_array_1::arrayToString][offset 0x00000006][found address of value 'T_array_1'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : @getsym::Getch][offset 0x0000008E][found address of Byte][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : T_set_2::clear][offset 0x00000004] Instruction cannot be verified.
[IL]: Error: [plzero.exe : T_alfa::arrayToString][offset 0x00000006][found address of value 'T_alfa'][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [plzero.exe : T_symset::clear][offset 0x00000003] Instruction cannot be verified.
[IL]: Error: [plzero.exe : T_array_15::arrayToString][offset 0x00000005][found address of value 'T_array_15'][expected unmanaged pointer] Unexpected type on the stack.
17 Error(s) Verifying plzero.exe
Exit code: 1

running the executable
-----------------------------------------------------

Exit code: 0