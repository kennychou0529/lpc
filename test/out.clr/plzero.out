    0 var x, squ;
    1  
    1 procedure square;
    1 begin
    3    squ := x * x
    5 end;
    0  jmp  1    0
    1  jmp  1    2
    2  int  1    3
    3  lod  1    3
    4  lod  1    3
    5  opr  1    4
    6  sto  1    4
    7  opr  1    0
    8  
    8 begin
    9    x := 1;
   11    while x < 10 do
   15    begin
   15       call square;
   16       x := x + 1
   18    end
   20 end.
    0  jmp  1    8
    1  jmp  1    2
    2  int  1    3
    3  lod  1    3
    4  lod  1    3
    5  opr  1    4
    6  sto  1    4
    7  opr  1    0
    8  int  1    5
    9  lit  1    1
   10  sto  1    3
   11  lod  1    3
   12  lit  1   10
   13  opr  1   10
   14  jpc  1   21
   15  cal  1    2
   16  lod  1    3
   17  lit  1    1
   18  opr  1    2
   19  sto  1    3
   20  jmp  1   11
   21  opr  1    0
 start pl/0
1
1
2
4
3
9
4
16
5
25
6
36
7
49
8
64
9
81
10
 end pl/0
